packages/react-components/src/components/ChatMessage/ChatMessageActionsFlyout.tsx

```diff
@@ -32,34 +41,93 @@ export interface ChatMessageActionFlyoutProps {
  * @private
  */
 export const ChatMessageActionFlyout = (props: ChatMessageActionFlyoutProps): JSX.Element => {
-  const menuItems = useMemo(
-    (): IContextualMenuItem[] => [
-      {
-        key: 'Edit',
-        text: props.strings.editMessage,
-        itemProps: { styles: props.increaseFlyoutItemSize ? menuItemIncreasedSizeStyles : undefined },
-        iconProps: { iconName: 'MessageEdit', styles: menuIconStyleSet },
-        onClick: props.onEditClick
-      },
-      {
-        key: 'Remove',
-        text: props.strings.removeMessage,
-        itemProps: { styles: props.increaseFlyoutItemSize ? menuItemIncreasedSizeStyles : undefined },
-        iconProps: {
-          iconName: 'MessageRemove',
-          styles: menuIconStyleSet
+  const theme = useTheme();
+  const menuItems = useMemo((): IContextualMenuItem[] => {
+    // only show read by x of x if more than 3 participants in total including myself
+    if (
+      props.participantCountNotIncludingSelf &&
+      props.messageThreadReadCount !== undefined &&
+      props.participantCountNotIncludingSelf >= 2 &&
+      props.strings.messageReadCount
```

Comment:
We assume that strings always exist. So checking for `props.strings.messageReadCount` isn't necessary / looks off.

Reply:
I'm checking for props.strings.messageReadCount because it is an optional param

Reply:
Ah, new strings that we add need to be optional... because of backwards compatibility.

@mgamis-msft can correct me here. Don't we fallback to our own en-us strings for anything that Contoso doesn't provide? So there is _no way_ for Contoso to initialize the composite in a way where strings are missing?

If this is the case we should:
* Make sure all strings are *required*.
* Only require `Partial<LocalType>` in our constructors and provider exported in the API.

For this PR, @carocao-msft could you add a TODO once Miguel resolves this thread with the right way ahead. You are right that you don't have any option given the types at present.

Reply:
updated